api/app/controllers/webhook.controller.ts

```diff
@@ -154,3 +156,81 @@ export async function create(req: Request, res: Response) {
     }
   }
 }
+
+export async function handlePushEvent(req: Request, res: Response) {
+  if (!verify_signature(req)) {
+    return res.status(401).json("Unauthorized");
+  }
+
+  const pushEvent = req.body;
+  if (!pushEvent) {
+    return res.status(500).json({
+      message: "No push event found in the request body.",
+    });
+  }
+
+  const commits = pushEvent.commits;
+  if (!commits) {
+    return res.status(500).json({
+      message: "No commits found in the request body.",
+    });
+  }
+
+  let responseStatus = 200;
+  let responseMessage = "success";
+
+  try {
+    for (const commit of commits) {
+      const changedFiles = [...commit.added, ...commit.modified];
+      for (const file of changedFiles) {
```

Comment:
At this point, we are iterating over all the commits and over all the changed files (new and modified).
Based on the `if` checks and the `break`statements, my understanding is that if we have multiple commits or multiple files per commit, as soon as one of the files is not added, we will get an error response, even if the first file was added successfully. Also, because of that, the rest of the files (for that commit) will not be processed.

So, in a scenario where I push 3 commits with one file each (or 1 commit with 3 files), I will get a response based on the file that it was not added successfully.

Reply:
I get your point, this works based on my limited knowledge of what tstbtc processes per time. On the assumption that you can only process and commit one file(transcript) per request. If that is not the case I can modify it to handle multiple transcripts(a file in a commit is a transcript) and save the data in db. But I will advise we keep it at one file per commit. Then we can track modified files when in review

Also for queueing, I think we should only track added files in the commits.

Reply:
Good point, let's keep it one transcript per commit for now.

We will care for modified files later, but for now you can remove that also.